### ISSUE TITLE
Boost Serialization unit test failing in 10.2.X

### ISSUE THREAD

**Issue by smuzaffar** [cms-sw/cmssw#28812]
@ggovi , We have one unit test failing in 10.2.X IBs [a] . Unit test is trying to read data of 10.6.X which has boost version 1.67. If this is not supported then can we fix the test to ignore such failures? 

[a]
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc6_amd64_gcc700/CMSSW_10_2_X_2020-01-28-1100/unitTestLogs/CondTools/Ecal#/63509
```
----- Begin Fatal Exception 28-Jan-2020 11:56:02 CET-----------------------
An exception of category 'ConditionDatabase' occurred while
   [0] Calling endJob for module ExTestEcalDAQAnalyzer/'conf_o2o'
Exception Message:
Payload of type EcalCondTowerObjectContainer&lt;EcalDAQStatusCode&gt; with id 0b9bf5ead2562ef3f9de32c1a222e19c3c7005d5 could not be loaded. An exception of category 'ConditionDatabase' occurred.
Exception Message:
De-serialization failed: the current boost version (1_63) is unable to read the payload. Data might have been serialized with an incompatible version. Payload serialization info:  {
"CMSSW_version": "CMSSW_10_6_0",
"architecture": "slc7_amd64_gcc700",
"technology": "boost/serialization",
"tech_version": "1_67"
 }
 from default_deserialize 
 from Session::fetchPayload 
----- End Fatal Exception -------------------------------------------------
```

**Comment by makortel** [comment]
assign db

**Comment by ggovi** [comment]
@smuzaffar 
Not sure how to proceed here. The test is fetching the last payload available, that in the present case has been recently stored during the MWGR with 10_6_0. That means, in 10_2 we will be never able to execute it properly. Shall we maybe disable the test?

**Comment by davidlange6** [comment]
Sounds like the test should be changed as to be fully reproducible (Use a fixed payload)




On 29 Jan 2020, at 13:59, ggovi <notifications@github.com> wrote:

﻿

@smuzaffar<https://github.com/smuzaffar>
Not sure how to proceed here. The test is fetching the last payload available, that in the present case has been recently stored during the MWGR with 10_6_0. That means, in 10_2 we will be never able to execute it properly. Shall we maybe disable the test?

**Comment by ggovi** [comment]
@davidlange6 
ok i'll work along this line...

**Comment by fabiocos** [comment]
@smuzaffar @ggovi this failure is there since a while, and we had already discussed it, as this is "known and understood" I did not worry much about it, also accounting for the fact that 10_2_X is basically frozen as far as DB developments are concerned. Anyway if we want to have a clean situation I believe that @davidlange6 suggestion is the way to go.

**Comment by davidlange6** [comment]
On Jan 30, 2020, at 12:20 PM, Fabio Cossutti <notifications@github.com<mailto:notifications@github.com>> wrote:


@smuzaffar<https://github.com/smuzaffar> @ggovi<https://github.com/ggovi> this failure is there since a while, and we had already discussed it, as this is "known and understood" I did not worry much about it, also accounting for the fact that 10_2_X is basically frozen as far as DB developments are concerned. Anyway if we want to have a clean situation I believe that @davidlange6<https://github.com/davidlange6> suggestion is the way to go.


Its more a basic design idea - our release tests should be reproducible..this includes the conditions payloads that they use.  Hard to debug if things change from under you.

**Comment by ggovi** [comment]
@davidlange6 
I need to check with the Ecal experts, since in these tests we are currently reading from the OMDS production, involving a content that - even in the case of a frozen input payload - will be evolving with time ( storing online data ).

**Comment by tvami** [comment]
Hi @smuzaffar since 10_2_X is closed, can this issue be closed?

**Comment by smuzaffar** [comment]
10.2.X is not really closed .. it is still open for bug fixes and we are regularly building IBs (11.2.X is closed though). But if we do not care much for this unit test in 10.2.X then I can close this issue

**Comment by tvami** [comment]
+db
 * resolved in https://github.com/cms-sw/cmssw/pull/35517
 * intestingly I wrote ```resolves https://github.com/cms-sw/cmssw/issues/28812``` in the description of that PR but it didnt close this issue
